framework/helpers/BaseHtml.php

```diff
@@ -974,27 +976,32 @@ public static function radioList($name, $selection = null, $items = [], $options
             $index++;
         }
 
-        $separator = isset($options['separator']) ? $options['separator'] : "\n";
+        if (isset($options['separator'])) {
+            $separator = $options['separator'];
+            unset($options['separator']);
+        } else {
+            $separator = "\n";
+        }
+
         if (isset($options['unselect'])) {
             // add a hidden field so that if the list box has no option being selected, it still submits a value
             $hidden = static::hiddenInput($name, $options['unselect']);
+            unset($options['unselect']);
         } else {
             $hidden = '';
         }
 
-        if (isset($options['tag']) && $options['tag'] == false) {
-            $tag = false;
-        } else {
-            $tag = isset($options['tag']) ? $options['tag'] : 'div';
+        if (!isset($options['tag'])) {
+            return $hidden . static::tag('div', implode($separator, $lines), $options);
         }
 
-        unset($options['tag'], $options['unselect'], $options['encode'], $options['separator'], $options['item'], $options['itemOptions']);
-
-        if ($tag === false) {
+        if ($options['tag'] == false) {
             return $hidden . implode($separator, $lines);
-        } else {
-            return $hidden . static::tag($tag, implode($separator, $lines), $options);
         }
+
+        $tag = $options['tag'];
+        unset($options['tag']);
+        return $hidden . static::tag($tag, implode($separator, $lines), $options);
```

Comment:
I think we can store `implode($separator, $lines)` in some variable not to duplicate it 3 times